From 4916c7fef3fdf3fa3cb296589febeb8a20a3507f Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Wed, 06 Jul 2022 08:45:13 +0300
Subject: [PATCH] [SW-94323] gaudi2: fix bug in clearing recovery flag.

Instead of clearing recovery flag, flag is being set
in erom recovery hence after firmware update the flow
always goes to erom recovery Therefore correcting it.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Idb1cc98c08fadd2027b5ea6b72febda09de2996b
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index d3aed0c..4547e4f 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1094,7 +1094,7 @@
 	struct gaudi2_cold_rst_data cold_rst_data;
 
 	cold_rst_data.data = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-	cold_rst_data.recovery_flag = 1;
+	cold_rst_data.recovery_flag = 0;
 	regmap_writel(cold_rst_data.data, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
 }
 
